src/metabase/models/permissions.clj

```diff
@@ -1132,22 +1157,38 @@
     (update-fn group-id db-id new-perms)
     (throw (ee-permissions-exception perm-type))))
 
+(defn update-global-execution-permission
+  "Set the global execution permission (\"/execute/\") for the group
+  with ID `group-id` to `new-perms`."
+  [group-id new-perms]
+  (when-not (or (= group-id (:id (perms-group/all-users)))
+                (premium-features/has-feature? :advanced-permissions))
+    (throw (ee-permissions-exception :execute)))
+  (delete-related-permissions! group-id "/execute/")
+  (when (= new-perms :all)
+    (grant-permissions! group-id "/execute/")))
+
 (s/defn ^:private update-group-permissions!
   [group-id :- su/IntGreaterThanZero new-group-perms :- StrictDBPermissionsGraph]
   (doseq [[db-id new-db-perms] new-group-perms]
-    (doseq [[perm-type new-perms] new-db-perms]
-      (case perm-type
-        :data
-        (update-db-data-access-permissions! group-id db-id new-perms)
+    (if (= db-id :execute)
```

Comment:
This is quite ugly. Is renaming `db-id` to something like `db-id-or-execute` enough?